src/mappings/tokens.ts

```diff
@@ -51,36 +52,56 @@ export function getTokenById(tokenId: i32): Token {
 }
 
 export function _createToken(id: string, address: Address, timestamp: BigInt, txHash: Bytes): Token {
-  log.info('[createToken] Create Token {} with address {}', [id, address.toHex()])
+  let addressString = address.toHex()
+  log.info('[createToken] Create Token {} with address {}', [id, addressString])
 
   // Create token
   let token = new Token(id)
   token.address = address
   token.fromBatchId = epochToBatchId(timestamp)
 
-  // Add symbol (optional)
-  let erc20 = Erc20.bind(address)
-  let symbolAttempt = erc20.try_symbol()
-  if (symbolAttempt.reverted) {
-    log.warning('Adding a ERC20 token with no "symbol". Address: {}', [address.toHex()])
-    token.symbol = null
-  } else {
-    token.symbol = symbolAttempt.value
-  }
+  var deprecatedToken: TokenDetails | null = DEPRECATED_TOKENS.has(addressString)
+    ? DEPRECATED_TOKENS.get(addressString)
+    : null
```

Comment:
This is assembly script so it might not apply, but on TS when you do a Map<T>.get() the return type is T|undefined, so you might not need to check it exists first.
Not the case?

Reply:
nup, unfortunately https://www.assemblyscript.org/stdlib/map.html